### ISSUE TITLE
Sorting does not work correctly when autodocs is set to 'tags'

### ISSUE THREAD

**Issue by SpaNb4** [prantlf/storybook-multilevel-sort#12]
When `autodocs` uses the default value ('tags'), groups with stories that have `tags: ['autodocs']` are sorted above other groups, ignoring the order specified in `storyOrder`.

**To reproduce:**
Make the following changes:
`Header.stories.js`:
![image](https://github.com/prantlf/storybook-multilevel-sort/assets/69504277/a102e8d3-84a5-467d-acf9-64ae7b35e6e2)

`Main.js`
![image](https://github.com/prantlf/storybook-multilevel-sort/assets/69504277/d90fc676-372c-4c99-a886-c83b42c5701c)

**Expected result**: 
Articles
Elements
Components

**Actual result**:
Articles
Components
Elements

**Comment by prantlf** [comment]
I looked at one of my productive storybooks which I migrated to Storybook 8 recently and found out why I didn't encounter this problem - i disabled `autodocs` :-)

```diff
-  docs: {
-    autodocs: 'tag'
-  },
+  docs: {},
```

I'll try figuring out how to fix it.

**Comment by prantlf** [comment]
Yes, I can reproduce it. Strange. It looks like that ordering by the story type has something to do with it. Except for `storyOrder`, it's possible to specify `typeOrder` too. But the autogenerated "Docs" articles work the best at the top and it's probably never changed:

```js
configureSort({
  storyOrder: {
    ...
  },
  // default - autogenerated Docs at first, then hand-written stories
  typeOrder: ['docs', 'story']
})
```

If I stop ordering by the story type and keep only ordering by the story name, the root level of story titles gets fixed, but, as expected, the autogenerated Docs end up sorted by name:

![obrazek](https://github.com/user-attachments/assets/2ffea784-e744-435b-aaf8-350fd534e9a5)

The picture above can be achieved by setting an empty array to `typeOrder`:

```js
configureSort({
  storyOrder: {
    ...
  },
  typeOrder: []
})
```

I looks like that the if the `docs` stories aren't included in each group, the sorting doesn't work well. I wonder where lies the rug.

**Comment by prantlf** [comment]
I figured it out. Storybook compares pairs of stories across all levels. With the default configuration `typeOrder: ['docs', 'story']`, I considered a story of the type `docs` to _always_ come before a story of the type `story`. Always, even if the group names should be sorted by the configuration (or alphabetically) the other way round. For example:

    Group:  Elements/Button  Components/Navigation/Header
    Name:   Active           Docs
    Type:   story            docs
    First:                   * (because docs < story)

The story `Elements/Button/Active` should come before `Components/Navigation/Header/Docs`, because the example sets `Elements` before `Components` on the first level of the `sortOrder` object. But the code in the comparison doesn't care of the group name, because right after comparing the story types and finding their orders not the same, the function bails out.

The fix will be considering the `typeOrder` only if both stories are in the same group. Otherwise only `storyOrder` will control the ordering.

**Comment by prantlf** [comment]
Opps, it looks like that putting "Fixes #12" to the commit message automatically closes the issue. Well, let's try to keep it. If there's still something wrong, you can reopen it.

**Comment by prantlf** [comment]
Thank you for your report! It helped me to discover a bug in the sorting. The fix is available in [2.0.1](https://github.com/prantlf/storybook-multilevel-sort/releases/tag/v2.0.1). Enjoy! :-)
